tests/robustness/validate/operations.go

```diff
@@ -52,45 +50,50 @@ func validateLinearizableOperationsAndVisualize(lg *zap.Logger, operations []por
 	}
 }
 
-func validateSerializableOperations(t *testing.T, lg *zap.Logger, operations []porcupine.Operation, persistedRequests []model.EtcdRequest) {
+func validateSerializableOperations(lg *zap.Logger, operations []porcupine.Operation, replay *model.EtcdReplay) (lastErr error) {
 	lg.Info("Validating serializable operations")
-	staleReads := filterSerializableReads(operations)
-	if len(staleReads) == 0 {
-		return
-	}
-	sort.Slice(staleReads, func(i, j int) bool {
-		return staleReads[i].Input.(model.EtcdRequest).Range.Revision < staleReads[j].Input.(model.EtcdRequest).Range.Revision
-	})
-	replay := model.NewReplay(persistedRequests)
-	for _, read := range staleReads {
+	for _, read := range operations {
 		request := read.Input.(model.EtcdRequest)
 		response := read.Output.(model.MaybeEtcdResponse)
-		validateSerializableOperation(t, replay, request, response)
+		err := validateSerializableRead(lg, replay, request, response)
+		if err != nil {
+			lastErr = err
+		}
 	}
+	return lastErr
 }
 
-func filterSerializableReads(operations []porcupine.Operation) []porcupine.Operation {
+func filterSerializableOperations(clients []report.ClientReport) []porcupine.Operation {
 	resp := []porcupine.Operation{}
-	for _, op := range operations {
-		request := op.Input.(model.EtcdRequest)
-		if request.Type == model.Range && request.Range.Revision != 0 {
-			resp = append(resp, op)
+	for _, client := range clients {
+		for _, op := range client.KeyValue {
+			request := op.Input.(model.EtcdRequest)
+			if request.Type == model.Range && request.Range.Revision != 0 {
+				resp = append(resp, op)
+			}
```

Comment:
Just a question: iiuc, all range requests are linearized by default, so:
(1) how are we doing serialized ranges? I see how we implement them in our model, but not sure of the distinction.
(2) how is this `if` condition telling us that an operation is a serializable operation?